### DISCUSSION TITLE
使用合约完成批量转账问题

### DISCUSSION THREAD

**Discussion by v1xingyue** [creatorsand/move-co-learn#32] [Q&A] [answered]
转账部分链接：

https://github.com/v1xingyue/sui-shell/blob/abac75563ff67ffc16a09b622d2a013f5dc2eb5c/src/components/MyPool.tsx#L112
使用 TransactionData 循环 调用多个moveCall 

合约部分: 
https://github.com/v1xingyue/sui-shell/blob/abac75563ff67ffc16a09b622d2a013f5dc2eb5c/move/sui-shell/sources/coin_pool.move#L38

报错信息如下： 

The transaction inputs contain duplicated ObjectRef's | (RpcError:-5000) (code: -5000)

![image](https://github.com/CreatorsDAO/move-co-learn/assets/974169/65b02cd3-9307-4059-9057-b50d444f0d10)

具体大家可以使用 https://sui-shell.vercel.app/ 测试。 错误可稳定复现。

**Comment by ShirleneLiu** [comment]
只有星sir在认真学习的世界达成了

**Reply by v1xingyue** [reply]
我要点个赞，并采纳为答案么。 😁

**Comment by v1xingyue** [comment, marked as answer]
类型传递错误，需要传递 object的时候，需要使用 tx.object(objectID) 来操作。 循环部分代码改为: 

```typescript
transferItems.forEach((item) => {
      tx.moveCall({
        target: CallTarget("coin_pool", "withdraw_to_address") as any,
        typeArguments: [coinType],
        arguments: [
          tx.pure(item.amount),
          tx.pure(item.address),
          tx.object(operatePool.id),  // 需要使用 tx.object 
        ],
      });
    });
```
